li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -228,6 +236,20 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
       // prepare to handle large messages.
       _largeMessageEnabled = configs.getBoolean(LiKafkaProducerConfig.LARGE_MESSAGE_ENABLED_CONFIG);
 
+      _encryptionEnabled = configs.getBoolean(LiKafkaProducerConfig.ENCRYPTION_ENABLED_CONFIG);
+      if (_encryptionEnabled) {
+        if (topicEncrypterDecrypterManager == null) {
+          LOG.warn("Cannot detect customized {}. The default one will be used.",
+              TopicEncrypterDecrypterManager.class.getSimpleName());
+          _topicEncrypterDecrypterManager = configs.getConfiguredInstance(TOPIC_ENCRYPTION_MANAGER_CLASS_CONFIG,
+              TopicEncrypterDecrypterManager.class);
+        } else {
+          _topicEncrypterDecrypterManager = topicEncrypterDecrypterManager;
```

Comment:
shouldn't the else part throw a configuration missing exception? If `_encryptionEnabled` is true, it should always have a `TopicEncrypterDecrypterManager` configured. In the existing code, I can simply pass `null` in the constructor for `TopicEncrypterDecrypter` and enable encryption. It will throw NPE down the code.

Reply:
The reason for such design uses the idea of LM support. I want to allocate a default de/encrypter for users who pass null in the constructor for TopicEncrypterDecrypter and enable encryption. Not so sure about the cause of NPE as I have added a check for `null`  TopicEncrypterDecrypter.